components/style/selector_impl.rs

```diff
@@ -8,6 +8,73 @@ use selectors::parser::{ParserContext, SelectorImpl};
 pub enum PseudoElement {
     Before,
     After,
+    FirstLine,
+
+    // https://mxr.mozilla.org/mozilla-central/source/layout/style/nsCSSAnonBoxList.h
```

Comment:
Why is all thing going here rather than in a separate `SelectorImpl` impl in `ports/geckolib`? Wasn’t that the point of making the parser generic?